### DISCUSSION TITLE
ScrollArea: child with flex display not taking 100% height

### DISCUSSION THREAD

**Discussion by Nikhil-Vats** [mantinedev/mantine#5826] [Q&A]
I am using ScrollArea element like this - 
```javascript
<ScrollArea  
  className="grow h-full"
  classNames={{
    root: "grow h-full",
    viewport: "flex",
  }}>
  <Accordion className="flex h-full grow>
     <child1>
     <child2>
     <child3 className="grow"> // I want this accordion item (textarea) to take all empty space in the page so that there is no whitespace at end
     <child4>
  </Accordion>
</ScrollArea?  
``` 

Because the deepest element of ScrollArea component is a `div` with `display: table`, the flex is not growing to take up empty space. The only fix I could find was to add `height: 100%` to the deepest `div` with `display: table` but there is no selector/class for it, I tried it using devtools. Is there a way for a child of ScrollArea to grow to take all empty space on page without using a hack like adding `height: 100%` without a selector since I am using tailwind and not SCSS.

**Comment by rtivital** [comment]
Add a div element inside scroll area that will be a flex container

**Reply by Nikhil-Vats** [reply]
I have tried that but it is not working.

The image below shows that scrollarea has 100% height
<img width="1160" alt="Screenshot 2024-02-28 at 12 06 10 AM" src="https://github.com/mantinedev/mantine/assets/31413407/3d3460e3-6b56-4ca4-88ea-d1422db93e8f">
I have a flex container inside scrollarea which doesn't take parent's height with flex-grow, 100% height and other properties.
<img width="1155" alt="image" src="https://github.com/mantinedev/mantine/assets/31413407/fc076ed3-d908-48c6-987f-62493bfa9af2">

**Comment by tombohub** [comment]
Tailwind added support for styling the children: https://tailwindcss.com/blog/tailwindcss-v3-4#style-children-with-the-variant

**Comment by Hikari-54** [comment]
I'm facing the same problem. 
Solved it through SCSS. 
Add this class to ScrollArea and div with display: table will take 100% of height.

```
.scrollAreaFix {
  height: 100%;

  > * {
    height: 100%;

    > * {
      height: 100%;
    }
  }
}
```

After you can hang space-between on the internal Flex.

Other methods have not solved the problem so far.

**Reply by HaTrang96** [reply]
It seems this method is a bit tricky, and unfortunately, it doesn’t work with Firefox

**Comment by msruan** [comment]
Did you resolve it?  I needed to use just a div with _overflow-auto_ instead scrollarea and it works.
Also in my case adding _height: 100%_ on the scrollarea children not worked.
